calculator.rb

```diff
@@ -0,0 +1,131 @@
+def val_operation(operation)
+  valid = false
+  while valid == false
+    if ["add", "+", "subtract", "-", "multiply", "*", "divide", "/", "exponent", "^", "modulo", "%"].include? operation
+      valid = true
+    else
+      puts ""
+      puts "That is not a valid operator. Please try again."
+      print "Please choose one operator(name or symbol): "
+      operation = gets.chomp.downcase
+    end
+  end
+  return operation
+end
+
+# Operand validation function.
+# Checking that the input is a valid number.
+# Will print error message and prompt for a new input until a valid one is obtained.
+# Will also return either int or float as appropriate.
+def val_number(number, prompt)
+  valid = false
+
+  # Function determines if the user entered an int or float by checking
+  # for the prescence of a decimal point.
+  # Converts string input into appropriate class and returns that value.
+  # If...the number is a float
+  if number.match?(/\.+/)
+    while valid == false
+      converted = number.to_f.to_s
+      if converted == number
+        valid = true
+      else
+        puts ""
+        puts "That is not a number. Please try again."
+        print prompt
+        number = gets.chomp
+      end
+    end
+    return number.to_f
+    # Else...the number is an int
+  else
+    while valid == false
+      converted = number.to_i.to_s
+      if converted == number
+        valid = true
+      else
+        puts ""
+        puts "That is not a number. Please try again."
+        print prompt
+        number = gets.chomp
+      end
+    end
+    return number.to_i
+  end
+end
+
+puts `clear`
+
+# Gives users options to select from.
+puts "Welcome to the Calculator program! Which operator would you like to choose?"
+puts "1. add(+)"
+puts "2. subtract(-)"
+puts "3. multiply(*)"
+puts "4. divide(/)"
+puts "5. exponent(^)"
+puts "6. modulo(%)"
+
+puts ""
+
+# Collects user input for operator.
+# User input goes through operator validation function.
+print "Please choose one operator(name or symbol): "
+operation = gets.chomp.downcase
+operation = val_operation(operation)
+
+puts ""
+
+# Collects user input for the first operand.
+# User input goes through operand validation function.
+print "Please enter a number for the math operation: "
+operand_1 = gets.chomp
+operand_1 = val_number(operand_1, "Please enter a number for the math operation: ")
+
+puts ""
+
+# Collects user input for the second operand.
+# User input goes through operand validation function.
+print "Please enter a second number for the math operation: "
+operand_2 = gets.chomp
+operand_2 = val_number(operand_2, "Please enter a second number for the math operation: ")
+
+# In cases of division or modulus, second operand cannot be zero.
+# Validation function for operands does not check for this case.
+# Code below will print error message and prompt for a new input
+# until a valid non-zero value is obtained.
+if ["divide", "/", "modulo", "%"].include? operation
```

Comment:
Consider putting this functionality in a method similar to what you did for checking for a valid number.